RemoveBoolVariation recipe with methodPattern for other SDK(Other than LaunchDrakly) is throwing TraitErrors

<!--
Thank you for reporting an issue with OpenRewrite!
We appreciate you taking the time to help us improve.
Please fill out the template below to help us understand and reproduce the issue.
Feel free to delete any sections that don't apply to your issue.
-->

## What version of OpenRewrite are you using?
<!--
Whenever possible please try to replicate your issue with the latest versions of OpenRewrite.
The latest major and minor versions of OpenRewrite projects are usually listed here:
https://docs.openrewrite.org/reference/latest-versions-of-every-openrewrite-module
For patch releases check the GitHub Releases page for the respective project.

We release every few weeks, so it's possible that your issue has already been fixed.

If you want to try the most recent changes that haven't been fully released yet, you can check out our snapshot releases
https://docs.openrewrite.org/reference/snapshot-instructions
-->
I am using
- Java v17
- OpenRewrite v6.14.0
- Maven/Gradle plugin v6.14.0
- rewrite-launchdrakly v0.5.0

## How are you running OpenRewrite?
<!--
Are you using the Maven plugin, Gradle plugin, Moderne CLI, Moderne.io or something else?
Is your project a single module or a multi-module project?

Can you share your configuration so that we can rule out any configuration issues?

Is your project public? If so, can you share a link to it?
Code snippets can also be shared privately via [our public Slack](https://join.slack.com/t/rewriteoss/shared_invite/zt-nj42n3ea-b~62rIHzb3Vo0E1APKCXEA).
-->
I am using the Gradle plugin, and my project is a multi module project.
```xml
plugins {
    id 'org.springframework.boot' version '3.0.4'
    id 'io.spring.dependency-management' version '1.1.0'
    id 'java'
    id 'org.openrewrite.rewrite' version '6.14.0'
}
```

## What is the smallest, simplest way to reproduce the problem?
<!--
Sometimes the logs indicate a recipe stumbled over a particular pattern of code.
If you can share a code snippet that reproduces the issue, that will help us fix it faster.
We also accept [pull requests that merely replicate an issue](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes), as a step up to a full fix.

A code snippet can be something simple like this, or similar for other languages:
-->
```java
public class ToggleChecker {
	public ToggleChecker() {
		
	}
	
	public  boolean isToggleEnabled(String toggleName,boolean fallback) {
		Map<String,Boolean> toggleMap=new  HashMap<>();
		toggleMap.put("FEATURE_TOGGLE1", true);
		toggleMap.put("FEATURE_TOGGLE2", true);
		toggleMap.put("FEATURE_TOGGLE3", false);
		
		return toggleMap.containsKey(toggleName);
	}
	

}
//Service class
public class ServiceImpl {
	private ToggleChecker toggleChecker=new ToggleChecker();
	public  void methodA() {
		if(toggleChecker.isToggleEnabled("FEATURE_TOGGLE1",false)) {
		  LOG.info("Feature Toggle  enabled"); 
	  }else {
		  LOG.info("Feature Toggle Not enabled");
	  }
	}
	

}
```
rewrite.yml
``` yml

type: specs.openrewrite.org/v1beta/recipe
name: com.osd.RemoveBoolVariationExample
displayName: Remove `isToggleEnabled` for feature key example
recipeList:
  - org.openrewrite.launchdarkly.RemoveBoolVariation:
      featureKey: FEATURE_TOGGLE1
      replacementValue: true
      methodPattern: com.osd.util.ToggleChecker isToggleEnabled(String,boolean)

```
## What did you expect to see?
<!-- A code snippet, or a description of the behavior you expected helps us write a test to ensure the issue is fixed. -->
```java
public class ServiceImpl {
	private ToggleChecker toggleChecker=new ToggleChecker();
	public  void methodA() {
		  LOG.info("Feature Toggle  enabled"); 
	
	}
	
```

## What did you see instead?
<!-- A code snippet, or a description of the behavior you saw instead of the above expected result. -->
```java
Exception
```

## What is the full stack trace of any errors you encountered?
<!-- When errors occur, please include the output of `--stacktrace` for Gradle or `--debug` for Maven. -->
```
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':microservices:product-service:rewriteDryRun'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:142)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:282)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:140)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:128)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:69)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:322)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:309)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:302)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:288)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:462)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:379)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:49)
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: Error while visiting com/services/ServiceImpl.java: org.openrewrite.analysis.trait.util.TraitErrorsException: TraitErrors: 
        - Method must be created from class org.openrewrite.java.tree.J$MethodDeclaration but was class org.openrewrite.java.tree.J$Lambda
        - No parent Method found
        - Field must be declared in a class, interface, or anonymous class
  org.openrewrite.analysis.trait.util.TraitErrors.doThrow(TraitErrors.java:43)
  fj.data.Either$RightProjection.on(Either.java:531)
  fj.data.Validation.on(Validation.java:146)
  org.openrewrite.analysis.trait.expr.VarAccessBase$1.visitVariable(VarAccess.java:187)
  org.openrewrite.analysis.trait.expr.VarAccessBase$1.visitVariable(VarAccess.java:175)
  org.openrewrite.java.tree.J$VariableDeclarations$NamedVariable.acceptJava(J.java:5955)
  org.openrewrite.java.tree.J.accept(J.java:59)
  org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
  org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:324)
  org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1369)
  org.openrewrite.java.JavaVisitor.lambda$visitVariableDeclarations$29(JavaVisitor.java:963)
  org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
  org.openrewrite.java.JavaVisitor.visitVariableDeclarations(JavaVisitor.java:963)
  org.openrewrite.java.tree.J$VariableDeclarations.acceptJava(J.java:5840)
  org.openrewrite.java.tree.J.accept(J.java:59)
  org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
  ...
        at org.openrewrite.gradle.DelegatingProjectParser.unwrapInvocationException(DelegatingProjectParser.java:163)
        at org.openrewrite.gradle.DelegatingProjectParser.dryRun(DelegatingProjectParser.java:113)
        at org.openrewrite.gradle.RewriteDryRunTask.run(RewriteDryRunTask.java:47)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:125)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
        at org.gradle.api.internal.tasks.execution.TaskExecution$3.run(TaskExecution.java:236)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
        at org.gradle.api.internal.tasks.execution.TaskExecution.executeAction(TaskExecution.java:221)
        at org.gradle.api.internal.tasks.execution.TaskExecution.executeActions(TaskExecution.java:204)
        at org.gradle.api.internal.tasks.execution.TaskExecution.executeWithPreviousOutputFiles(TaskExecution.java:187)
        at org.gradle.api.internal.tasks.execution.TaskExecution.execute(TaskExecution.java:165)
        at org.gradle.internal.execution.steps.ExecuteStep.executeInternal(ExecuteStep.java:89)
        at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:40)
        at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:53)
        at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:50)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:50)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:40)
        at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:68)
        at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:38)
        at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:41)
        at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:74)
        at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:55)
        at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:51)
        at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:29)
        at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.executeDelegateBroadcastingChanges(CaptureStateAfterExecutionStep.java:124)
        at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:80)
        at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:58)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:48)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:36)
        at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:181)
        at org.gradle.internal.execution.steps.BuildCacheStep.lambda$execute$1(BuildCacheStep.java:71)
        at org.gradle.internal.Either$Right.fold(Either.java:175)
        at org.gradle.internal.execution.caching.CachingState.fold(CachingState.java:59)
        at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:69)
        at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:47)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:36)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:25)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:36)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:22)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:110)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$2(SkipUpToDateStep.java:56)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:56)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:38)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:73)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:44)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:89)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:50)
        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:102)
        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:57)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:76)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:50)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.executeWithNoEmptySources(SkipEmptyWorkStep.java:254)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:91)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:56)
        at org.gradle.internal.execution.steps.RemoveUntrackedExecutionStateStep.execute(RemoveUntrackedExecutionStateStep.java:32)
        at org.gradle.internal.execution.steps.RemoveUntrackedExecutionStateStep.execute(RemoveUntrackedExecutionStateStep.java:21)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
        at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:43)
        at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:31)
        at org.gradle.internal.execution.steps.AssignWorkspaceStep.lambda$execute$0(AssignWorkspaceStep.java:40)
        at org.gradle.api.internal.tasks.execution.TaskExecution$4.withWorkspace(TaskExecution.java:281)
        at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:40)
        at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:30)
        at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:37)
        at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:27)
        at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:44)
        at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:33)
        at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:76)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:139)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:128)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:69)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:322)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:309)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:302)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:288)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:462)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:379)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:49)
Caused by: java.lang.RuntimeException: Error while visiting com/services/ServiceImpl.java: org.openrewrite.analysis.trait.util.TraitErrorsException: TraitErrors: 
        - Method must be created from class org.openrewrite.java.tree.J$MethodDeclaration but was class org.openrewrite.java.tree.J$Lambda
        - No parent Method found
        - Field must be declared in a class, interface, or anonymous class
  org.openrewrite.analysis.trait.util.TraitErrors.doThrow(TraitErrors.java:43)
  fj.data.Either$RightProjection.on(Either.java:531)
  fj.data.Validation.on(Validation.java:146)
  org.openrewrite.analysis.trait.expr.VarAccessBase$1.visitVariable(VarAccess.java:187)
  org.openrewrite.analysis.trait.expr.VarAccessBase$1.visitVariable(VarAccess.java:175)
  org.openrewrite.java.tree.J$VariableDeclarations$NamedVariable.acceptJava(J.java:5955)
  org.openrewrite.java.tree.J.accept(J.java:59)
  org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
  org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:324)
  org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1369)
  org.openrewrite.java.JavaVisitor.lambda$visitVariableDeclarations$29(JavaVisitor.java:963)
  org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
  org.openrewrite.java.JavaVisitor.visitVariableDeclarations(JavaVisitor.java:963)
  org.openrewrite.java.tree.J$VariableDeclarations.acceptJava(J.java:5840)
  org.openrewrite.java.tree.J.accept(J.java:59)
  org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
  ...
        at org.openrewrite.gradle.isolated.ResultsContainer$1.lambda$preVisit$0(ResultsContainer.java:115)
        at org.openrewrite.gradle.isolated.ResultsContainer$1.preVisit(ResultsContainer.java:112)
        at org.openrewrite.gradle.isolated.ResultsContainer$1.preVisit(ResultsContainer.java:108)
        at org.openrewrite.gradle.isolated.ResultsContainer_1_JavaVisitor.preVisit(ResultsContainer_1_JavaVisitor.zig:102)
        at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:247)
        at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:324)
        at org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1369)
        at org.openrewrite.java.JavaVisitor.lambda$visitVariableDeclarations$29(JavaVisitor.java:963)
        at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
        at org.openrewrite.java.JavaVisitor.visitVariableDeclarations(JavaVisitor.java:963)
        at org.openrewrite.java.tree.J$VariableDeclarations.acceptJava(J.java:5840)
        at org.openrewrite.java.tree.J.accept(J.java:59)
        at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
        at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:324)
        at org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1369)
        at org.openrewrite.java.JavaVisitor.lambda$visitLambda$17(JavaVisitor.java:795)
        at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
        at org.openrewrite.java.JavaVisitor.visitLambda(JavaVisitor.java:794)
        at org.openrewrite.java.tree.J$Lambda.acceptJava(J.java:3199)
        at org.openrewrite.java.tree.J.accept(J.java:59)
        at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
        at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:324)
        at org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1369)
        at org.openrewrite.java.JavaVisitor.lambda$visitContainer$37(JavaVisitor.java:1419)
        at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
        at org.openrewrite.java.JavaVisitor.visitContainer(JavaVisitor.java:1419)
        at org.openrewrite.java.JavaVisitor.visitMethodInvocation(JavaVisitor.java:922)
        at org.openrewrite.java.tree.J$MethodInvocation.acceptJava(J.java:3945)
        at org.openrewrite.java.tree.J.accept(J.java:59)
        at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
        at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:324)
        at org.openrewrite.java.JavaVisitor.visitLeftPadded(JavaVisitor.java:1397)
        at org.openrewrite.java.JavaVisitor.visitVariable(JavaVisitor.java:1303)
        at org.openrewrite.java.tree.J$VariableDeclarations$NamedVariable.acceptJava(J.java:5955)
        at org.openrewrite.java.tree.J.accept(J.java:59)
        at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
        at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:324)
        at org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1369)
        at org.openrewrite.java.JavaVisitor.lambda$visitVariableDeclarations$29(JavaVisitor.java:963)
        at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
        at org.openrewrite.java.JavaVisitor.visitVariableDeclarations(JavaVisitor.java:963)
        at org.openrewrite.java.tree.J$VariableDeclarations.acceptJava(J.java:5840)
        at org.openrewrite.java.tree.J.accept(J.java:59)
        at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
        at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:324)
        at org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1369)
        at org.openrewrite.java.JavaVisitor.lambda$visitBlock$4(JavaVisitor.java:401)
        at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
        at org.openrewrite.java.JavaVisitor.visitBlock(JavaVisitor.java:400)
        at org.openrewrite.java.tree.J$Block.acceptJava(J.java:841)
        at org.openrewrite.java.tree.J.accept(J.java:59)
        at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
        at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:324)
        at org.openrewrite.java.JavaVisitor.visitMethodDeclaration(JavaVisitor.java:883)
        at org.openrewrite.java.tree.J$MethodDeclaration.acceptJava(J.java:3685)
        at org.openrewrite.java.tree.J.accept(J.java:59)
        at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
        at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:324)
        at org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1369)
        at org.openrewrite.java.JavaVisitor.lambda$visitBlock$4(JavaVisitor.java:401)
        at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
        at org.openrewrite.java.JavaVisitor.visitBlock(JavaVisitor.java:400)
        at org.openrewrite.java.tree.J$Block.acceptJava(J.java:841)
        at org.openrewrite.java.tree.J.accept(J.java:59)
        at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
        at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:324)
        at org.openrewrite.java.JavaVisitor.visitClassDeclaration(JavaVisitor.java:488)
        at org.openrewrite.java.tree.J$ClassDeclaration.acceptJava(J.java:1290)
        at org.openrewrite.java.tree.J.accept(J.java:59)
        at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
        at org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:324)
        at org.openrewrite.java.JavaVisitor.lambda$visitCompilationUnit$10(JavaVisitor.java:501)
        at org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
        at org.openrewrite.java.JavaVisitor.visitCompilationUnit(JavaVisitor.java:501)
        at org.openrewrite.java.tree.J$CompilationUnit.acceptJava(J.java:1568)
        at org.openrewrite.java.tree.J.accept(J.java:59)
        at org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
        at org.openrewrite.gradle.isolated.ResultsContainer.getRecipeErrors(ResultsContainer.java:119)
        at org.openrewrite.gradle.isolated.ResultsContainer.getFirstException(ResultsContainer.java:99)
        at org.openrewrite.gradle.isolated.DefaultProjectParser.dryRun(DefaultProjectParser.java:344)
        at org.openrewrite.gradle.isolated.DefaultProjectParser.dryRun(DefaultProjectParser.java:338)
        at org.openrewrite.gradle.DelegatingProjectParser.lambda$dryRun$3(DelegatingProjectParser.java:114)
        at org.openrewrite.gradle.DelegatingProjectParser.unwrapInvocationException(DelegatingProjectParser.java:155)
```

## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?
<!-- Indicate if this is something you would like to work on, and how we can best support you in doing so. -->

--- a/src/main/java/org/openrewrite/launchdarkly/ChangeVariationDefault.java
+++ b/src/main/java/org/openrewrite/launchdarkly/ChangeVariationDefault.java
@@ -69,12 +69,7 @@ public class ChangeVariationDefault extends Recipe {
             public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
                 J.MethodInvocation mi = super.visitMethodInvocation(method, ctx);
                 Expression firstArgument = mi.getArguments().get(0);
-                boolean isFirstArgumentFeatureKey =
-                        CursorUtil
-                                .findCursorForTree(getCursor(), firstArgument)
-                                .bind(c -> ConstantFold.findConstantLiteralValue(c, String.class))
-                                .map(featureKey::equals)
-                                .orSome(false);
+                boolean isFirstArgumentFeatureKey = isFeatureKey(firstArgument);
                 Expression lastArgument = mi.getArguments().get(mi.getArguments().size() - 1);
                 if (BOOL_VARIATION_MATCHER.matches(mi) && isFirstArgumentFeatureKey) {
                     return changeValue(mi, lastArgument, new J.Literal(Tree.randomId(), Space.SINGLE_SPACE, Markers.EMPTY, defaultValue, defaultValue, null, JavaType.Primitive.Boolean));
@@ -89,6 +84,13 @@ public class ChangeVariationDefault extends Recipe {
                     return changeValue(mi, lastArgument, new J.Literal(Tree.randomId(), Space.SINGLE_SPACE, Markers.EMPTY, defaultValue, defaultValue, null, JavaType.Primitive.Double));
                 }
                 return mi;
+            }
+
+            private Boolean isFeatureKey(Expression firstArgument) {
+                return J.Literal.isLiteralValue(firstArgument, featureKey) || CursorUtil.findCursorForTree(getCursor(), firstArgument)
+                        .bind(c -> ConstantFold.findConstantLiteralValue(c, String.class))
+                        .map(featureKey::equals)
+                        .orSome(false);
             }
 
             private J.MethodInvocation changeValue(J.MethodInvocation mi, Expression existingValue, J.Literal newValue) {

--- a/src/main/java/org/openrewrite/launchdarkly/ChangeVariationDefault.java
+++ b/src/main/java/org/openrewrite/launchdarkly/ChangeVariationDefault.java
@@ -69,25 +69,24 @@ public class ChangeVariationDefault extends Recipe {
             public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
                 J.MethodInvocation mi = super.visitMethodInvocation(method, ctx);
                 Expression firstArgument = mi.getArguments().get(0);
-                boolean isFirstArgumentFeatureKey = isFeatureKey(firstArgument);
                 Expression lastArgument = mi.getArguments().get(mi.getArguments().size() - 1);
-                if (BOOL_VARIATION_MATCHER.matches(mi) && isFirstArgumentFeatureKey) {
+                if (BOOL_VARIATION_MATCHER.matches(mi) && isFeatureKey(firstArgument)) {
                     return changeValue(mi, lastArgument, new J.Literal(Tree.randomId(), Space.SINGLE_SPACE, Markers.EMPTY, defaultValue, defaultValue, null, JavaType.Primitive.Boolean));
                 }
-                if (STRING_VARIATION_MATCHER.matches(mi) && isFirstArgumentFeatureKey) {
+                if (STRING_VARIATION_MATCHER.matches(mi) && isFeatureKey(firstArgument)) {
                     return changeValue(mi, lastArgument, new J.Literal(Tree.randomId(), Space.SINGLE_SPACE, Markers.EMPTY, defaultValue, "\"" + defaultValue + "\"", null, JavaType.Primitive.String));
                 }
-                if (INT_VARIATION_MATCHER.matches(mi) && isFirstArgumentFeatureKey) {
+                if (INT_VARIATION_MATCHER.matches(mi) && isFeatureKey(firstArgument)) {
                     return changeValue(mi, lastArgument, new J.Literal(Tree.randomId(), Space.SINGLE_SPACE, Markers.EMPTY, defaultValue, defaultValue, null, JavaType.Primitive.Int));
                 }
-                if (DOUBLE_VARIATION_MATCHER.matches(mi) && isFirstArgumentFeatureKey) {
+                if (DOUBLE_VARIATION_MATCHER.matches(mi) && isFeatureKey(firstArgument)) {
                     return changeValue(mi, lastArgument, new J.Literal(Tree.randomId(), Space.SINGLE_SPACE, Markers.EMPTY, defaultValue, defaultValue, null, JavaType.Primitive.Double));
                 }
                 return mi;
             }
 
             private Boolean isFeatureKey(Expression firstArgument) {
-                return J.Literal.isLiteralValue(firstArgument, featureKey) || CursorUtil.findCursorForTree(getCursor(), firstArgument)
+                return CursorUtil.findCursorForTree(getCursor(), firstArgument)
                         .bind(c -> ConstantFold.findConstantLiteralValue(c, String.class))
                         .map(featureKey::equals)
                         .orSome(false);

--- a/src/main/java/org/openrewrite/launchdarkly/RemoveBoolVariation.java
+++ b/src/main/java/org/openrewrite/launchdarkly/RemoveBoolVariation.java
@@ -25,6 +25,7 @@ import org.openrewrite.internal.lang.Nullable;
 import org.openrewrite.java.JavaVisitor;
 import org.openrewrite.java.MethodMatcher;
 import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.Expression;
 import org.openrewrite.java.tree.J;
 import org.openrewrite.java.tree.JavaType;
 import org.openrewrite.java.tree.Space;
@@ -77,19 +78,21 @@ public class RemoveBoolVariation extends Recipe {
             @Override
             public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
                 J.MethodInvocation mi = (J.MethodInvocation) super.visitMethodInvocation(method, ctx);
-                boolean isFirstArgumentFeatureKey =
-                        CursorUtil
-                                .findCursorForTree(getCursor(), mi.getArguments().get(0))
-                                .bind(c -> ConstantFold.findConstantLiteralValue(c, String.class))
-                                .map(featureKey::equals)
-                                .orSome(false);
-                if (methodMatcher.matches(mi) && isFirstArgumentFeatureKey) {
+                if (methodMatcher.matches(mi) && isFeatureKey(mi.getArguments().get(0))) {
                     doAfterVisit(new SimplifyConstantIfBranchExecution().getVisitor());
                     doAfterVisit(new RemoveUnusedLocalVariables(null).getVisitor());
                     doAfterVisit(new RemoveUnusedPrivateFields().getVisitor());
                     return new J.Literal(Tree.randomId(), Space.SINGLE_SPACE, Markers.EMPTY, replacementValue, String.valueOf(replacementValue), null, JavaType.Primitive.Boolean);
                 }
                 return mi;
+            }
+
+            private boolean isFeatureKey(Expression firstArgument) {
+                return J.Literal.isLiteralValue(firstArgument, featureKey) ||
+                       CursorUtil.findCursorForTree(getCursor(), firstArgument)
+                               .bind(c -> ConstantFold.findConstantLiteralValue(c, String.class))
+                               .map(featureKey::equals)
+                               .orSome(false);
             }
         };
         return Preconditions.check(new UsesMethod<>(methodMatcher), visitor);

--- a/src/main/java/org/openrewrite/launchdarkly/RemoveBoolVariation.java
+++ b/src/main/java/org/openrewrite/launchdarkly/RemoveBoolVariation.java
@@ -88,8 +88,7 @@ public class RemoveBoolVariation extends Recipe {
             }
 
             private boolean isFeatureKey(Expression firstArgument) {
-                return J.Literal.isLiteralValue(firstArgument, featureKey) ||
-                       CursorUtil.findCursorForTree(getCursor(), firstArgument)
+                return CursorUtil.findCursorForTree(getCursor(), firstArgument)
                                .bind(c -> ConstantFold.findConstantLiteralValue(c, String.class))
                                .map(featureKey::equals)
                                .orSome(false);

--- a/src/test/java/org/openrewrite/launchdarkly/RemoveBoolVariationTest.java
+++ b/src/test/java/org/openrewrite/launchdarkly/RemoveBoolVariationTest.java
@@ -18,9 +18,11 @@ package org.openrewrite.launchdarkly;
 import org.junit.jupiter.api.Test;
 import org.openrewrite.DocumentExample;
 import org.openrewrite.InMemoryExecutionContext;
+import org.openrewrite.Issue;
 import org.openrewrite.java.JavaParser;
 import org.openrewrite.test.RecipeSpec;
 import org.openrewrite.test.RewriteTest;
+import org.openrewrite.test.SourceSpec;
 
 import static org.openrewrite.java.Assertions.java;
 
@@ -80,7 +82,7 @@ class RemoveBoolVariationTest implements RewriteTest {
               
               class Foo {
                   private static final String FEATURE_FLAG_123ABC = "flag-key-123abc";
-                  
+              
                   private LDClient client = new LDClient("sdk-key-123abc");
                   void bar() {
                       LDContext context = null;
@@ -297,10 +299,10 @@ class RemoveBoolVariationTest implements RewriteTest {
           java(
             """
               package com.acme.bank;
-                            
+              
               import com.launchdarkly.sdk.LDContext;
               import com.launchdarkly.sdk.server.LDClient;
-                            
+              
               public class CustomLaunchDarklyWrapper {
                   private LDClient client = new LDClient("sdk-key-123abc");
                   public boolean featureFlagEnabled(String key, boolean fallback) {
@@ -318,6 +320,60 @@ class RemoveBoolVariationTest implements RewriteTest {
                   private CustomLaunchDarklyWrapper wrapper = new CustomLaunchDarklyWrapper();
                   void bar() {
                       if (wrapper.featureFlagEnabled("flag-key-123abc", false)) {
+                          // Application code to show the feature
+                          System.out.println("Feature is on");
+                      }
+                      else {
+                        // The code to run if the feature is off
+                          System.out.println("Feature is off");
+                      }
+                  }
+              }
+              """,
+            """
+              class Foo {
+                  void bar() {
+                      // Application code to show the feature
+                      System.out.println("Feature is on");
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    @Issue("https://github.com/openrewrite/rewrite-launchdarkly/issues/23")
+    void customMethodPatternNoConstants() {
+        RemoveBoolVariation featureToggle1 = new RemoveBoolVariation("FEATURE_TOGGLE1", true, "com.osd.util.ToggleChecker isToggleEnabled(String, boolean)");
+        // language=java
+        rewriteRun(
+          spec -> spec.recipe(featureToggle1),
+          java(
+            """
+            package com.osd.util;
+            import java.util.Map;
+            import java.util.HashMap;
+
+            public class ToggleChecker {
+                public boolean isToggleEnabled(String toggleName, boolean fallback) {
+                    Map<String,Boolean> toggleMap = new HashMap<>();
+                    toggleMap.put("FEATURE_TOGGLE1", true);
+                    toggleMap.put("FEATURE_TOGGLE2", true);
+                    toggleMap.put("FEATURE_TOGGLE3", false);
+                    return toggleMap.containsKey(toggleName);
+                }
+            }
+            """,
+            SourceSpec::skip
+          ),
+          java(
+            """
+              import com.osd.util.ToggleChecker;
+              class Foo {
+                  private ToggleChecker checker = new ToggleChecker();
+                  void bar() {
+                      if (checker.isToggleEnabled("FEATURE_TOGGLE1", false)) {
                           // Application code to show the feature
                           System.out.println("Feature is on");
                       }

